query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusRestClient.java

```diff
@@ -0,0 +1,122 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import lombok.NoArgsConstructor;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+public class PrometheusRestClient {
+  private static final String INSTANT_QUERY = "api/v1/query";
+  private static final String RANGE_QUERY = "api/v1/query_range";
+
+  private String host;
+  private int port;
+  private OkHttpClient okHttpClient;
+
+  public PrometheusRestClient(String host, int port, OkHttpClient okHttpClient) {
+    this.okHttpClient = okHttpClient;
+    this.host = host;
+    this.port = port;
+  }
+
+  public List<ImmutablePair<Request, PromQLMetricResponse>> execute(PromQLQuery query) {
```

Comment:
Why return a list of pairs instead of a map?

Reply:
Here also, I was thinking, while building query service response might need to iterate and would be easy. But, move to the map. If, requires, will move to LinkedHashMap.